### DISCUSSION TITLE
How to Perform Global Search Within Files in NvChad?

### DISCUSSION THREAD

**Discussion by RuiOkazaki** [NvChad/NvChad#2420] [Q&A] [answered]
Hello everyone,

I've been using NvChad and I'm really enjoying the experience so far. However, I'm having trouble figuring out how to perform a global search within files. I've looked through the documentation and searched online, but I haven't found a clear answer.

Could anyone guide me on how to achieve this? Is there a specific plugin or command that I should be using for this functionality in NvChad?

Thank you in advance for your help!

Best regards,

Rui Okazaki

**Comment by siduck** [comment]
leader ff @RuiOkazaki 

leader is space key in nvchad

do leader ch, to know all mappings

**Reply by RuiOkazaki** [reply]
Hello @siduck,

First of all, thank you for creating and maintaining NvChad; it's been a fantastic experience using it. I apologize for the lack of clarity in my initial question.

What I'm actually looking to do is search for a specific string across all files in my project. I know about the space key being the leader key in NvChad and have tried leader ch to view all mappings, but I'm still not sure how to perform a global search within files.

Could you please guide me on how to accomplish this?

Thank you once again for your time and assistance.

Best regards

**Reply by siduck** [reply, marked as answer]
@RuiOkazaki leader fw

**Reply by RuiOkazaki** [reply]
@siduck 
I just wanted to update you all that I've managed to solve my issue by installing ripgrep. Thank you so much for your guidance and support!

**Reply by siduck** [reply]
@RuiOkazaki always read the docs

![image](https://github.com/NvChad/NvChad/assets/59060246/2cd9f766-2bef-4911-993f-c60cad14611c)

**Reply by RuiOkazaki** [reply]
I've read through the documentation and I must say it's been incredibly helpful! Even as someone who's not very familiar with Neovim, I was able to install plugins and change key mappings successfully. Thank you so much for making this so accessible!
